### ISSUE TITLE
📚 `noUndeclaredDependencies` and `useImportType` report issues with output from TypeScript code actions

### ISSUE THREAD

**Issue by chrisgrieser** [biomejs/website#50]
### Documentation URL

https://biomejs.dev/linter/rules/use-import-restrictions/

### Description

Using typescript code actions `Add missing imports` results in code that is reported by biome. I am using the default typescript language server settings. 

```ts
// biome reports `noUndeclaredDependencies`
import MyModule from "src/main"; // <- this line is auto-generated by "add missing imports"

// biome does not report`noUndeclaredDependencies` when changed to this:
import MyModule from "../main";
```

```ts
// biome reports `useImportType`
import MyModule from "module"; // <- this line is auto-generated by "add missing imports"
// …
let mod: MyModule;
```

While I think the first can be resolved by changing the typescript configuration to `importModuleSpecifierPreference = relative` ([see here](https://github.com/typescript-language-server/typescript-language-server/blob/master/docs/configuration.md)), ~~I am not aware of any typescript setting to deal with the latter.~~ (`preferTypeOnlyAutoImports` does that, see comment below)

### Expectations

That a recommended rule (`useImportType`) reports output of typescript as an issue is quite irritating, and probably very confusing for less experienced users.

You could argue that this is somewhat a bug of those rules, but technically, the rules work as expected. Possible solutions to this:

#### A) Change the documentation
1. Mention that these rules oppose the output of typescript code actions with default settings.
2. Explain which typescript settings need to be changed to make the output of the code actions comply with biome's rules.

~~However, 2. is complicated by the fact that there is (to my knowledge) no typescript setting to make typescript add type imports instead of imports.~~ (`preferTypeOnlyAutoImports` does that, as commented below)

Also, changing typescript configs is not always straightforward. Since tsserver is not "fully" open, users with an editor other than VSCode need to install special language server replacements to be able to configure the `typescript.preferences`, e.g. [typescript tools](https://github.com/pmizio/typescript-tools.nvim).

That means: Quite a lot of effort to be able to comply with biome for new users.

#### B) Change rule behavior
Alternatively, the rule's behavior should be modified to be less strict. `noUndeclaredDependencies` could be modified to check if an import is simply project-relative, before reporting a violation.

Not sure how to deal with `useImportType` though. I'd consider demoting it from a recommended to a purely opt-in rule.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
@Conaclos I moved this issue here, but maybe I made a mistake. What do we need to do to address the issue and close it?

**Comment by Conaclos** [comment]
> What do we need to do to address the issue and close it?

We could add a note in the docs of `useImportType` about the vscode config `typescript.preferences.preferTypeOnlyAutoImports`.

Regarding `noUndeclaredDependencies` I am not sure what to do because the use of paths relative to the project root seems pretty rare? I don't know if there is an official doc about that.

**Comment by Conaclos** [comment]
@chrisgrieser Could you share with us your current tsconfig file that allows to use import relative to the project root? This will help us to document this.

**Comment by chrisgrieser** [comment]
@Conaclos tbh, part of the reason why I opened this issue was because I wasn't totally sure on what all the configs are, that you need to change to make biome "compatible" with typescript. And I have to admit that for now, I have simply disabled the respective biome rules 🙈

**Comment by Sec-ant** [comment]
I think the import specifier is controlled by this option. Maybe we should add it to the document? This is an LSP option:

https://stackoverflow.com/a/53137571
